accounts/models.py

```diff
@@ -1,27 +1,74 @@
 from django.db import models
-from django.db.models.signals import post_save
-from django.contrib.auth.models import User
-from django.dispatch import receiver
+from django.contrib.auth.models import AbstractBaseUser,\
+                                       PermissionsMixin,\
+                                       BaseUserManager
 
 from transactions.models import Wallet, Transaction
 
 
-class Profile(models.Model):
-    user = models.OneToOneField(User,
-                                related_name='profile',
-                                on_delete=models.CASCADE)
+class UserProfileManager(BaseUserManager):
+    """Helps Django work with our custom user model"""
+
+    def create_user(self, email, first_name, last_name, password):
+        """Create a new user profile object"""
+
+        if not email:
+            raise ValueError('Users must have an email address.')
+        email = self.normalize_email(email)
+        user = self.model(email=email,
+                          first_name=first_name,
+                          last_name=last_name)
+
+        user.set_password(password)
+        user.save(using=self.db)
+        return user
+
+    def create_superuser(self, email, first_name, last_name, password):
+        """Creates and saves a new superuser with given details"""
+
+        user = self.create_user(email, first_name, last_name, password)
+        user.is_superuser = True
+        user.is_staff = True
+        user.save(using=self.db)
+        return user
+
+    def get_by_natural_key(self, email_):
+        print(email_)
```

Comment:
print? :)

Reply:
posiłkowałem się tym artykułem: https://medium.com/agatha-codes/options-objects-customizing-the-django-user-model-6d42b3e971a4 :) nie wiem czy to jest najlepsze rozwiązanie